File: golang-http-server/internal/db/models/attachment.go
package models

import (
	"database/sql"
	"time"
)

type Attachment struct {
	ID         string         `json:"id"`
	TaskID     string         `json:"task_id"`
	FileName   string         `json:"file_name"`
	FileKey    string         `json:"file_key"`
	MimeType   string         `json:"mime_type"`
	FileSize   int            `json:"file_size"`
	UploadedBy sql.NullString `json:"uploaded_by"`
	CreatedAt  time.Time      `json:"created_at"`
}

File: fastapi/src/db/models/attachment.py
from datetime import datetime
import uuid

from sqlalchemy import DateTime, ForeignKey, Integer, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.db.base import Base


class Attachment(Base):
    __tablename__ = "attachments"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    task_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("tasks.id", ondelete="cascade"),
        nullable=False,
    )
    file_name: Mapped[str] = mapped_column(String(255), nullable=False)
    file_key: Mapped[str] = mapped_column(String(512), nullable=False)
    mime_type: Mapped[str] = mapped_column(String(128), nullable=False)
    file_size: Mapped[int] = mapped_column(Integer, nullable=False)
    uploaded_by: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("user.id", ondelete="set null")
    )
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

    task: Mapped[Task] = relationship(back_populates="attachments")
    uploader: Mapped[User] = relationship()
